changes to which image and waiting for couchdb to be ready

newtewt opened the pull request:
using the 2.x.x official [couch image](https://hub.docker.com/_/couchdb) vs the [apache repo image](https://hub.docker.com/r/apache/couchdb), added bash script to check that couch is responding before trying to move on setting up, switch from semicolon to ampersand so if the setup fails it fails grunt 

# Description

[description]

medic/medic#[number]

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

SCdF commented on Gruntfile.js:
```diff
@@ -493,19 +493,19 @@ module.exports = function(grunt) {
         cmd: [
-          `docker run -d -p 4984:5984 -p 4986:5986 --name e2e-couchdb --mount type=tmpfs,destination=/opt/couchdb/data apache/couchdb:latest`,
-          'sleep 1',
+          `docker run -d -p 4984:5984 -p 4986:5986 --name e2e-couchdb --mount type=tmpfs,destination=/opt/couchdb/data couchdb:2`,
```
You should drop the tmpfs stuff since it doesn't work on MacOS. We can add it back just for linux later once we determine it's worth it

newtewt replied:
cool I've removed. I tested on mac os but not windows. It seems the run command works but disregards the tmpfs flag.

newtewt commented:
@SCdF  I went ahead and moved it into it's own script and updated for the changes to removing `tmpfs` and add --rm to remove the container when it is shutdown.

SCdF commented on scripts/wait_for_couch.sh:
```diff
@@ -0,0 +1,6 @@
+count=0
```
can you either rename this to make it clear this is only for the e2e couchdb (because of 4984) or have the port passed as a parameter. I worry that months from now it will be confusing without one of those two changes

dianabarsan replied:
Agree, this would be my only comment as well.
Maybe we could put our `e2e` stuff under a single folder, so far it would be this script + `scripts/e2e-servers.js`.
